src/community/features-templating/features-templating-rest/src/test/java/org/geoserver/featurestemplating/rest/TemplateRestControllerTest.java

```diff
@@ -420,5 +430,57 @@ private void newTemplateInfo(
         info.setFeatureType(featureType);
         TemplateInfoDAO.get().saveOrUpdate(info);
         TemplateFileManager.get().saveTemplateFile(info, rawTemplate);
+        return info;
+    }
+
+    @Test
+    public void testUpdatedCacheOnPut() throws Exception {
```

Comment:
I would have expected this test to pass even before the changes... aren't the templates reloaded based on the last modification time? But I guess one has to wait up to a second to rely on that mechanism, while here we have an instant effect?

That said, I'm still confused... the title says "templating cache might retain orphan templates" and the description says "the template is moved from the feature type folder to the ws one" but this test only checks an update in place.